### ISSUE TITLE
Add obstruction detection directly into prop_testchamber_door entity

### ISSUE THREAD

**Issue by vrad-exe** [CitadelCore/Portal-2-Enhanced#741]
### Which component should be improved?

Entity

### Describe your suggestion

`prop_testchamber_door`s currently will always close when sent a `Close` input, ignoring any players or other objects which might be in the way, or forcibly killing players in coop. Most mappers will add triggers to the doors to handle this situation (either by waiting for the door to be unblocked, or crushing things in the way), but considering that test chamber doors are already their own entity, it'd be nice to have this integrated into it directly instead of needing to be manually added by the mapper.

### Expected result

An "Obstruction Behavior" keyvalue would be added to the entity, determining how the door behaves when sent a close input. It would have 3 settings:
- **Wait**: Keep the door open until the obstructions are removed. This would be the default mode in Hammer.
- **Destroy**: Close immediately and destroy the obstructions, killing players and fizzling objects.
- **Nothing**: Ignore players or objects obstructing the door, and close anyway. This would be used if the keyvalue wasn't present on the entity (for backwards compatibility with existing maps).

For actually detecting things obstructing the door, you could probably generate an appropriately-sized trigger set to oriented bounding box collisions (so diagonal doors work correctly) in code. That's basically what floor buttons do with the `trigger_portal_button` entity. Or perhaps just tracing inside the door frame would be sufficient

**Comment by Luke18033** [comment]
Perhaps also add a "push out" mode, which uses the same logic as light bridges to push obstructing players/objects out to one side.
